plugins/oauth2authenticator/src/Vanilla/Authenticator/OAuth2Authenticator.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+namespace Vanilla\Authenticator;
+
+use Garden\Web\Exception\ServerException;
+use Garden\Web\RequestInterface;
+use Vanilla\Models\SSOData;
+
+class OAuth2Authenticator extends SSOAuthenticator {
+
+    /** @var array */
+    public $providerData;
+
+    /**
+     * OAuth2Authenticator constructor.
+     *
+     * @param string $authenticatorID
+     * @param \Gdn_AuthenticationProviderModel $authProviderModel
+     */
+    public function __construct(
+        $authenticatorID,
+        \Gdn_AuthenticationProviderModel $authProviderModel
+    ) {
+        if (empty($authenticatorID)) {
+            $authenticatorID = 'oauth2';
+        }
+
+        parent::__construct($authenticatorID);
+
+        $this->providerData = $authProviderModel->getProviderByKey($authenticatorID);
+
+        if (empty($this->providerData)) {
+            throw new Exception('Empty providerData in OAuth2Authenticator.');
+        }
+    }
+
+    /**
+     * @inheritDoc
```

Comment:
I believe the `@inheritDoc` tag is required to be wrapped in curly braces: `{@inheritDoc}`